src/olympia/addons/models.py

```diff
@@ -1271,18 +1271,24 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def is_incomplete(self):
-        # Don't hassle add-ons with only unlisted versions.
-        if not self.is_listed:
-            return False
-
+    def has_complete_metadata(self, for_channel=None):
+        return all(self.get_required_metadata(for_channel=for_channel))
+
+    def get_required_metadata(self, for_channel=None):
+        if not for_channel:
+            for_channel = (amo.RELEASE_CHANNEL_LISTED if
+                           self.has_listed_versions() else
+                           amo.RELEASE_CHANNEL_UNLISTED)
+        if for_channel == amo.RELEASE_CHANNEL_UNLISTED:
+            # Add-ons with only unlisted versions have no required metadata.
+            return []
         latest_version = self.find_latest_version(
             channel=amo.RELEASE_CHANNEL_LISTED)
-        return not (
-            self.all_categories and
-            self.summary and
-            (not latest_version or latest_version.license)
-        )
+        return [
+            self.all_categories,
+            self.summary,
+            (not latest_version or latest_version.license),
```

Comment:
It was already there, but what about just `latest_version and latest_version.license` ? I find it less confusing.